### PR TITLE
Updates to SAM template and CircleCi config

### PR DESCRIPTION
**Pull request by crissupb**
Update CircleCi to only copy to S3 when building the master branch Issue https://github.com/18F/identity-devops/issues/2973
Copy template to prod lambda s3 bucket 
Set function names instead of using SAM template default name

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -108,9 +141,25 @@ commands:
...
             aws s3 cp identity-idp-functions.template.yaml s3://$DEPLOY_S3_BUCKET/circleci/identity-idp-functions/$CIRCLE_BRANCH/
             aws s3 cp identity-idp-functions.zip s3://$DEPLOY_S3_BUCKET/circleci/identity-idp-functions/$CIRCLE_BRANCH/
 
+  copy-to-s3-prod:
+    steps:
+      - run:
+          name: "Copy packages to S3 prod"
+          command: |
+            cd source
+            aws s3 cp identity-infra-functions.template.yaml s3://$prod_DEPLOY_S3_BUCKET/circleci/identity-infra-functions/$CIRCLE_BRANCH/ --profile prod
```
Comment (zachmargolis) [inline .circleci/config.yml:160]: is the variable `$PROD_DEPLOY_S3_BUCKET` or `$prod_DEPLOY_S3_BUCKET`? I'm not used to seeing mixed casing like this

File: .circleci/config.yml [hunk 2]
```diff
@@ -28,7 +40,19 @@ jobs:
...
+    executor: ubuntu-2004
+    steps:
+      - brew-install
+      - aws-cli-configure
+      - aws-cli-configure-prod
+      - sam-cli-install
+      - checkout
+      - build-repo
+      - copy-to-s3
+      - copy-to-s3-prod
```
Comment (zachmargolis) [inline .circleci/config.yml:55]: Can we make sure the prod part only happens on the `master` branch?

**Review by zachmargolis — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (crissupb) [reply .circleci/config.yml:55]: The build-deploy workflow doesn't copy to S3 now for either prod or sandbox.  The build-deploy-copy workflow has a filter to only run when the branch is set to master.

Comment (zachmargolis) [reply .circleci/config.yml:55]: Ok great, thanks for confirming

**Pull request merged**